test/unit-test/CMakeLists.txt

```
@@ -1,89 +1,94 @@
-list(APPEND mock_include_list
-            ${HTTP_PARSER_DIR}
-        )
-#list the definitions of your mocks to control what to be included
-list(APPEND mock_define_list
-            ""
-       )
-
-# ================= Create the library under test here (edit) ==================
-
-# list the files you would like to test here
-list(APPEND real_source_files
-            ${HTTP_SOURCES}
-        )
-# list the directories the module under test includes
-list(APPEND real_include_directories
-            .
-            ${HTTP_INCLUDE_PUBLIC_DIRS}
-            ${HTTP_INCLUDE_PRIVATE_DIRS}
-        )
-
-# =====================  Create UnitTest Code here (edit)  =====================
-
-# list the directories your test needs to include
-list(APPEND test_include_directories
-            .
-            ${HTTP_INCLUDE_PUBLIC_DIRS}
-            ${HTTP_INCLUDE_PRIVATE_DIRS}
-            ${HTTP_INCLUDE_THIRD_PARTY_DIRS}
-        )
-
-# =============================  (end edit)  ===================================
-
-set(mock_name "http_parser_mock")
-set(real_name "${project_name}_real")
-
-create_mock_list(${mock_name}
-                "${mock_list}"
-                "${MODULE_ROOT_DIR}/tools/cmock/project.yml"
-                "${mock_include_list}"
-                "${mock_define_list}"
-        )
-
-create_real_library(${real_name}
-                    "${real_source_files}"
-                    "${real_include_directories}"
-                    "${mock_name}"
-        )
-
-list(APPEND utest_link_list
-            -l${mock_name}
-            lib${real_name}.a
-        )
-
-list(APPEND utest_dep_list
-            ${real_name}
-        )
-
-set(utest_name "${project_name}_utest")
-set(utest_source "${project_name}_utest.c")
-create_test(${utest_name}
-            ${utest_source}
-            "${utest_link_list}"
-            "${utest_dep_list}"
-            "${test_include_directories}"
-        )
-
-set(utest_name "${project_name}_send_utest")
-set(utest_source "${project_name}_send_utest.c")
-create_test(${utest_name}
-            ${utest_source}
-            "${utest_link_list}"
-            "${utest_dep_list}"
-            "${test_include_directories}"
-        )
+# Project information.
+cmake_minimum_required ( VERSION 3.13.0 )
+project ( "CoreHTTP unit test"
+          VERSION 1.0.0 
+          LANGUAGES C )
+
+# Allow the project to be organized into folders.
+set_property( GLOBAL PROPERTY USE_FOLDERS ON )
+
+# Use C90.
+set( CMAKE_C_STANDARD 90 )
+set( CMAKE_C_STANDARD_REQUIRED ON )
+
+# Do not allow in-source build.
+if( ${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR} )
+    message( FATAL_ERROR "In-source build is not allowed. Please build in a separate directory, such as ${PROJECT_SOURCE_DIR}/build." )
+endif()
+
+# Set global path variables.
+get_filename_component(__MODULE_ROOT_DIR "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE)
+set(MODULE_ROOT_DIR ${__MODULE_ROOT_DIR} CACHE INTERNAL "HTTP source root.")
```

Suggestion to avoid confusion with `source/` folder
```suggestion
set(MODULE_ROOT_DIR ${__MODULE_ROOT_DIR} CACHE INTERNAL "coreHTTP repository root.")
```

---

sure

---

I also suggested changing `source` -> `repository`

---

sure